src/server.rs

```diff
@@ -605,8 +649,18 @@ impl RollupBoostServer {
 
         let (l2_payload, builder_payload) = tokio::join!(l2_client_future, builder_client_future);
         let (payload, context) = match (builder_payload, l2_payload) {
-            (Ok(Some(builder)), _) => Ok((builder, PayloadSource::Builder)),
-            (_, Ok(l2)) => Ok((l2, PayloadSource::L2)),
+            (Ok(Some(builder)), Ok(l2)) => {
+                histogram!("block_building_gas_difference")
```

Comment:
these can be defined in `impl SpanProcessor for MetricsSpanProcessor {` on the end of the call?